benchmarking/evaluations/metrics/average_solo_status.py

```diff
@@ -0,0 +1,51 @@
+from collections import Counter
+from typing import Dict, List
+
+from api.models.team_set import TeamSet
+from benchmarking.evaluations.interfaces import TeamSetMetric
+
+
+class AverageSoloStatus(TeamSetMetric):
+    """
+    Calculate the average number of students having solo status in a team. A student is only considered as solo status
+    once, even if they have multiple minority attributes.
+
+    Params
+    ------
+    minority_groups: Dict[int, List[int]]
+        A dictionary where the key is the attribute_id and the value is a list of attribute_values that are considered
+        as minority.
+    """
+
+    def __init__(self, minority_groups_map: Dict[int, List[int]], *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.minority_groups_map = minority_groups_map
+
+        if len(minority_groups_map) == 0:
+            raise ValueError("minority_groups cannot be empty")
+
+    def calculate(self, team_set: TeamSet) -> float:
+        total_solo_status_students = set()
```

Comment:
nit;

naming this `total_solo_status_students` implies it is a count, when it isn't,a nd then variable below is named similarly, even though they store completely different types of data